src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticDriver.php

```diff
@@ -29,37 +26,36 @@ class StaticDriver implements Driver
     public function __construct(Driver $underlyingDriver)
     {
         $this->underlyingDriver = $underlyingDriver;
+        parent::__construct($underlyingDriver);
     }
 
     public function connect(array $params): DriverConnection
     {
-        if (!self::$keepStaticConnections) {
-            return $this->underlyingDriver->connect($params);
+        if (!self::isKeepStaticConnections()
+            || !isset($params['dama.keep_static'])
+            || !$params['dama.keep_static']
+        ) {
+            return parent::connect($params);
         }
 
         $key = sha1(json_encode($params));
 
         if (!isset(self::$connections[$key])) {
-            self::$connections[$key] = $this->underlyingDriver->connect($params);
+            self::$connections[$key] = parent::connect($params);
             self::$connections[$key]->beginTransaction();
         }
 
-        return new StaticConnection(self::$connections[$key]);
-    }
+        $connection = self::$connections[$key];
 
-    public function getSchemaManager(\Doctrine\DBAL\Connection $conn, AbstractPlatform $platform): AbstractSchemaManager
-    {
-        return $this->underlyingDriver->getSchemaManager($conn, $platform);
-    }
+        $platform = isset($params['serverVersion'])
```

Comment:
hi @dmaicher.
i think there is a bug here.
given config like
```yaml
when@test:
    doctrine:
        dbal:
            connections:
                default:
                    dbname_suffix: '_test%env(default::TEST_TOKEN)%'
                    platform_service: BindHQ\Bundle\TestsBundle\Doctrine\DBAL\Platforms\PostgreSQLPlatform
                    use_savepoints: true
```
my custom platform service is ignored and not used here.
I thnk we need to use `$params['platform']` if set?

Reply:
Oh indeed. Looks like I missed that case. Its deprecated and won't work with DBAL 4 though.

Are you up for creating a new issue for it + look into fixing it? :blush:

Reply:
yep, i'll fix it!
what has changed in dbal4?

Reply:
See https://github.com/doctrine/dbal/blob/62cce0a4486ce8757fa43a3f72d95a81b89da9e7/src/Connection.php#L203-L208

Reply:
ah right, i assume there is a change in doctrine bundle to support this...

Reply:
The `platform_service` option is deprecated on DoctrineBundle. I think there is no out-of-the box solution yet. Might involve creating a custom middleware + registering it.